src/DirectoryExtraFieldDisplay.php

```diff
@@ -286,6 +298,19 @@ public function preprocessFacetList(array &$variables) {
         $group_items[$entity->bundle()]['items'][$key] = $item;
       }
     }
+
+    // This is usually on a channel node. If so remove facets not active on
+    // channel.
+    if (($channel = $this->routeMatch->getParameter('node'))
+      && $channel instanceof NodeInterface
+      && $channel->bundle() == 'localgov_directory'
+    ) {
+      $active_facets = array_column($channel->localgov_directory_facets_enable->getValue(), 'target_id');
+    }
+    if ($active_facets) {
```

Comment:
Can this lead to a PHP warning if we are not on a localgov_directory node?

Reply:
Yep. That's one of the test fails. Should be updated with [31db4e5](https://github.com/localgovdrupal/localgov_directories/pull/270/commits/31db4e5a6f16dbf04f653926c29766ae8b0459e5)